### PR TITLE
Proxy: Retry after 5 seconds on everry Request Exception

### PR DESCRIPTION
**Pull request by aliefhooghe**
I am using the Disney+ addons mainly to watch some of _The Simpsons_ episodes. 

I have been encountering playback error since some months ago. The symptom is a brutal freeze of the video playback and a loop playback of the last 50 ms of sound. The video timestamp continue to increase in the player ui. It had become almost impossible to watch an entire episode (~20 min) without encountering the error.

 I saw in the logs that this error was correlated with `requests.ReadTimeout` exceptions. So I generalized the request retry exception handling to `request.RequestException` which is a common parent type of `requests.ReadTimeout` and the previously catched `ConnectionError`. I also disabled testing the exception message. 

 This did not fix the problem at first because the default request timeout is too high (~25 seconds on my linux) and does not allow the client to get data soon enough in case of a retry. I then set the timeout to 5 seconds and the retries count to 5 (to match the total 25 seconds timeout). This allow the data to be fetched fast enough even when a retry is done.

This as fixed the bug for me, I have not anymore playback trouble. By adding log I saw that at least one retry was done during each episode, without any perceptible consequences. 

Are these settings applicable to every use cases or setups ? (I am using a quite fast fiber connection).

Thanks you very much for your work, hopping this will be useful.

### CODE REVIEW

**Comment by matthuisman** [comment]
the timout it uses can be set in slyguy common settings
https://github.com/matthuisman/slyguy.addons/blob/master/script.module.slyguy/resources/settings.xml#L41
which gets passed to the proxy here:
https://github.com/matthuisman/slyguy.addons/blob/master/script.module.slyguy/resources/modules/slyguy/gui.py#L486

So I think for you, change that timeout to 5s

I don't want 5s timeout hardcoded as some users may legitimately have connections that take longer

**Comment by aliefhooghe** [comment]
I totally agree that the timeout should be passed by setting instead of being hardcoded, I will change the setting value for myself.
However it would be sufficient to fix the issue because the current condition for retry is too restrictive:
```python
except ConnectionError as e:
    if 'Connection aborted' not in str(e) or i == retries-1:
```
There would be no retry in case of `Timeout` exception because of the the exceptions hierarchy:
```python
class ConnectionError(RequestException):
class Timeout(RequestException):
class ReadTimeout(Timeout):
```

I suggest to retry for any RequestException.

File: script.module.slyguy/resources/lib/proxy.py [hunk 1]
```diff
@@ -1241,14 +1241,16 @@ def _proxy_request(self, method, url):
...
             log.debug('REQUEST OUT: {} ({})'.format(url, method.upper()))
             try:
-                response = self._session['session'].request(method=method, url=url, headers=self._headers, data=self._post_data, allow_redirects=False, stream=True)
-            except ConnectionError as e:
-                if 'Connection aborted' not in str(e) or i == retries-1:
+                response = self._session['session'].request(
+                    method=method, url=url, headers=self._headers, timeout=5,
+                    data=self._post_data, allow_redirects=False, stream=True)
+            except RequestException as e:
+                if i == retries-1:
```
[the remark this answers is not in the document]
Comment (aliefhooghe) [reply script.module.slyguy/resources/lib/proxy.py:1253]: We could keep this to decide whether to retry or not

**Comment by matthuisman** [comment]
new IA just pushed out:
https://github.com/xbmc/inputstream.adaptive/commit/65ef8940e299dddf4a430fbcb1adefad4855f055

"[AdaptiveStream] allow segment download retrys for VOD"

This may do what you need actually :)

**Pull request closed without merging**